contributors/design-proposals/encryption.md

```diff
@@ -0,0 +1,436 @@
+# Encryption
+
+## Abstract
+
+The scope of this proposal is to ensure that resources can be encrypted at the
+datastore layer with sufficient metadata support to enable integration with
+multiple encryption providers and key rotation. Encryption will be optional for
+any resource, but will be used by default for the Secret resource. Secrets are
+already protected in transit via TLS.
+
+Full disk encryption of the volumes storing etcd data is already expected as
+standard security hygiene.  Adding the proposed encryption at the datastore
+layer defends against malicious parties gaining access to:
+
+- The etcd API
```

Comment:
It does not.  Someone gaining write access to etcd is effectively root on the entire cluster, the same as any other core database.  Secrets are the least of the problems at that point.  Agree should be separated.